### PR TITLE
Feature/dpopp/timer driver

### PR DESCRIPTION
**Pull request by ddp5730**
- Implemented the Timer Driver
- Timer triggers interrupt with given period
- Can start, stop, and reload the timer.

### CODE REVIEW

File: src/dev/platform/f3xx/f302x8/Timerf302x8.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+void Timerf302x8::startTimer(void (*irqHandler)(TIM_HandleTypeDef *htim)) {
+    TIM_TypeDef *timerPeripheral = this->halTimer->Instance;
+    stopTimer();
+    timerInterruptHandlers[getTimerInterruptIndex(timerPeripheral)] = irqHandler;
+    initTimer(timerPeripheral, this->clockPeriod);
+}
+
+void Timerf302x8::stopTimer() {
+    HAL_TIM_Base_DeInit(this->halTimer);
```
Comment (ddp5730) [inline src/dev/platform/f3xx/f302x8/Timerf302x8.cpp:157]: Do we want to completely de-init the timer here?  Another function call is `HAL_TIM_Base_Stop_IT()` which just pauses the timer module.  Realistically the functionality appears to be the same either way.

Comment (cbolles) [reply src/dev/platform/f3xx/f302x8/Timerf302x8.cpp:157]: We don't have to complete de-init. I imagine this will be used like a "pause" for the timer.

**Review by cbolles — CHANGES_REQUESTED** [review]

**Comment by ddp5730** [comment]
I forgot to implement the one off timer.  Therefore converting to a draft.

File: src/dev/platform/f3xx/f302x8/Timerf302x8.cpp (hunk 1 above)
Comment (ddp5730) [reply src/dev/platform/f3xx/f302x8/Timerf302x8.cpp:157]: I agree.

**Review by cbolles — APPROVED** [review]
Looks good!

**Pull request merged**